pkg/loki/modules.go

```diff
@@ -321,6 +322,19 @@ func (t *Loki) initTenantConfigs() (_ services.Service, err error) {
 }
 
 func (t *Loki) initDistributor() (services.Service, error) {
+	var trackedTee distributor.TrackedTee
+	if t.Cfg.Ingester.KafkaIngestion.Enabled {
+		kafkaTee, err := kafka_tee.NewTee(t.Cfg.KafkaConfig, t.Cfg.MetricsNamespace, prometheus.DefaultRegisterer, util_log.Logger, t.partitionRing)
+		if err != nil {
+			return nil, err
+		}
+		if t.Cfg.Ingester.KafkaIngestion.Required {
```

Comment:
In fact what we need is a bit different, Kafka push once activated will always be required. 

What we want to instead is the ability disable push to ingesters via GRPC.

I think we should stop using the Tee and just incorporate the logic in the distributors.

Reply:
The migration will be like this:
- 1. Push to both
- 2. Push to kafka only .

Reply:
Do we want to retain the ability to async write to kafka for testing purposes? Thats where I thought we would want to stick with the Tee approach.

Reply:
No I think we're going all in.